src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/GCHandle.T.cs

```diff
@@ -0,0 +1,132 @@
+            set
+            {
+                IntPtr handle = _handle;
+                GCHandle.CheckUninitialized(handle);
+                GCHandle.InternalSet(handle, value);
+            }
+        }
+
+        /// <summary>
+        /// Returns a new <see cref="GCHandle{T}"/> object created from a handle to a managed object.
+        /// </summary>
+        /// <param name="value">An <see cref="IntPtr"/> handle to a managed object to create a <see cref="GCHandle{T}"/> object from.</param>
+        /// <returns>A new <see cref="GCHandle{T}"/> object that corresponds to the value parameter.</returns>
+        /// <remarks>
+        /// The <see cref="IntPtr"/> representation of <see cref="GCHandle{T}"/> is not
+        /// interchangable with <see cref="GCHandle"/>.
+        /// </remarks>
+        public static GCHandle<T> FromIntPtr(IntPtr value) => new GCHandle<T>(value);
+
+        /// <summary>
+        /// Returns the internal integer representation of a <see cref="GCHandle{T}"/> object.
+        /// </summary>
+        /// <param name="value">A <see cref="GCHandle{T}"/> object to retrieve an internal integer representation from.</param>
+        /// <returns>An <see cref="IntPtr"/> object that represents a <see cref="GCHandle{T}"/> object.</returns>
+        /// <remarks>
+        /// The <see cref="IntPtr"/> representation of <see cref="GCHandle{T}"/> is not
+        /// interchangable with <see cref="GCHandle"/>.
+        /// </remarks>
+        public static IntPtr ToIntPtr(GCHandle<T> value) => value._handle;
+
+        /// <summary>
+        /// Releases this <see cref="GCHandle{T}"/>.
+        /// </summary>
+        public void Dispose()
+        {
+            // Free the handle if it hasn't already been freed.
+            // Unlike GCHandle.Free, no thread safety is provided.
+            IntPtr handle = _handle;
+            if (handle != IntPtr.Zero)
+            {
+                _handle = IntPtr.Zero;
+                GCHandle.InternalFree(handle);
+            }
+        }
+
+        /// <summary>Indicates whether the current instance is equal to another instance of the same type.</summary>
+        /// <param name="obj">An instance to compare with this instance.</param>
+        /// <returns>true if the current instance is equal to the other instance; otherwise, false.</returns>
+        public override readonly bool Equals([NotNullWhen(true)] object? obj) => obj is GCHandle<T> handle && Equals(handle);
+
+        /// <summary>Indicates whether the current instance is equal to another instance of the same type.</summary>
+        /// <param name="other">An instance to compare with this instance.</param>
+        /// <returns>true if the current instance is equal to the other instance; otherwise, false.</returns>
+        public readonly bool Equals(GCHandle<T> other) => _handle == other._handle;
+
+        /// <summary>
+        /// Returns an identifier for the current <see cref="GCHandle{T}"/> object.
```

Comment:
Hashcode is an identifier?

Reply:
I was copying the documentation from `GCHandle`.

Reply:
The GCHandle documentation does not look right.

The typical documentation for GetHashCode says "Returns the hash code for the current instance.". We should do the same here.